### ISSUE TITLE
Cannot get screen capture of Java programs (background rendered as transparent, compositing issues)

### ISSUE THREAD

**Issue by amkearns-usgs** [XQuartz/XQuartz#219]
When capturing screen images from a java program running remotely, the background and some features of the GUI do not display correctly, showing up in the image preview as transparent instead of the proper theming color. Based on similar issues in XQuartz (see https://github.com/XQuartz/XQuartz/issues/24), this may be the result of an issue with MacOS. The XQuartz window displays properly, but sharing screenshots of data and programs shows these errors. When saving the captured image, the transparent portions become black (see attachment).

Note that this problem does not occur in 256-color mode, only in higher color depths.

Relevant versioning information:
Mac OS 11.2.5 (Big Sur) on 2019 (Intel) MacBook Pro
XQuartz 2.8.1

To replicate:
Launch a remote session (SSH) using -XY (XQuartz)
Run a java program in the remote session
Capture output of the java program (i.e., command+shift+4)

Again, while I can't demonstrate this easily, this issue is **only** present when performing screen capture and normal operation of these java programs works fine (i.e., no obvious display artifacts with GUI components).

This appears to be the same issue as $24. Response to that ticket (otherwise unaddressed to my knowledge, given this issue persists) called for an Apple feedback number; I have submitted one for this issue, FB9724784. 

See attached image for example (black parts of program render as transparent on screen capture preview popup)
<img width="1593" alt="Output of Java program on XQuartz" src="https://user-images.githubusercontent.com/23105449/139121766-c733f925-2311-4755-aad0-e063030e7cea.png">

**Comment by amkearns-usgs** [comment]
I have another, possibly more informative example of why these issues are a problem. Here's more-or-less a program we run should look like when screen-captured.

<img width="950" alt="Screen Shot 2021-11-03 at 1 29 55 PM" src="https://user-images.githubusercontent.com/23105449/140392483-7ed0b9b3-d7c1-47f9-8df8-8c2f7d2754ae.png">

To get a working screen capture, we have to do something to the Swing graphics instance like setXORMode() that stops 2D acceleration from working normally. This makes the program slower so we try to get rid of these kinds of calls.

However, when the 2D acceleration works 'normally':

<img width="937" alt="Screen Shot 2021-11-03 at 6 11 27 PM" src="https://user-images.githubusercontent.com/23105449/140392690-20527a03-cb67-4088-8f25-b9f40a459743.png">

Something about the XQuartz renderer does not want to display the plotted region in screenshots. The program is perfectly viewable in the window -- this is only an issue with screen capture.

Stranger still is when the application has a popup or other GUI components overlaying the main frame:
<img width="1342" alt="Screen Shot 2021-11-04 at 10 37 43 AM" src="https://user-images.githubusercontent.com/23105449/140393015-bf8cddea-ecaf-4197-ba03-be37f90de0e8.png">

The area where the popup has a shadow has visibility of the underlying components. Any window, even ones not part of the java program (i.e., terminal windows) can be placed over the window to make these components viewable in screen capture.

Because it's only screen captures (and, seemingly, only when graphics acceleration is available) that cause this problem I would presume this is an issue in XQuartz. Most attempts I've made to try to get the obscured components to display properly in XMAX fail. My presumption is that this is some sort of conflict between lightweight and heavyweight components (i.e., something about the main frame causes it to draw over the other graphics when it shouldn't be).

I've changed the title of this issue to better reflect the nature of this problem.
